app/src/main/ui/src/hooks/useFilter.ts

```diff
@@ -0,0 +1,46 @@
+import {useEffect, useReducer} from "react";
+import {QueryOptions} from "./useAPI";
+import {DEFAULT_PAGE_SIZE, calcPageStart} from "../lib/pagination";
+import {useSearchParams} from "react-router-dom";
+
+const PAGE_QUERY_PARAM = "p";
+const SEARCH_QUERY_PARAM = "q"
+
+type QueryOptionsReducerArgs = QueryOptions & {
+    force: boolean;
+}
+
+function updateQueryOptions(currentState: QueryOptions, newState: QueryOptionsReducerArgs): QueryOptions {
+    if (currentState.start !== newState.start || currentState.query !== newState.query || newState.force) {
+        return {start: newState.start, limit: newState.limit, query: newState.query}
+    }
+    return currentState;
+}
+
+export function useFilter(): [(_: string) => void, (_: number) => void, () => void, QueryOptions] {
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [page, query] = [+(searchParams.get(PAGE_QUERY_PARAM) ?? 1), searchParams.get(SEARCH_QUERY_PARAM) ?? ""]
```

Comment:
What does the `+` do?

Reply:
The `+` converts a string value to number.